UnicornEngine/video/include/unicorn/video/Graphics.hpp

```diff
@@ -0,0 +1,187 @@
+    Graphics& operator=(Graphics& other) = delete;
+    Graphics& operator=(Graphics&& other) = delete;
+
+    /** @brief  Initializes the graphics system
+    *
+    *  Ensures that window management system supports Vulkan
+    *
+    *  @param  whichDriver Driver, which will render our all graphics.
+    *
+    *  @return @c true if initialization was successful, @c false otherwise
+    */
+    bool Init(const DriverType& whichDriver);
+
+    /** @brief  Deinitializes the graphics system
+    *
+    *  Deinitializes all Renderer-Window pairs
+    */
+    void Deinit();
+
+    /** @brief  Graphics renderer loop */
+    bool Render();
+
+    /** @brief  Spawns new window
+    *
+    *  Also creates renderer context for created window
+    *
+    *  @param  width           desired width
+    *  @param  height          desired height
+    *  @param  name            desired name
+    *  @param  pMonitor        pointer to a monitor where window
+    *                          should be created
+    *  @param  pSharedWindow   pointer to another window that should
+    *                          share all resources with created window
+    *
+    *  @return a pointer to newly created window object
+    */
+    UNICORN_EXPORT system::Window* SpawnWindow(int32_t width,
+                                               int32_t height,
+                                               const std::string& name,
+                                               system::Monitor* pMonitor = nullptr,
+                                               system::Window* pSharedWindow = nullptr);
+
+    /** @brief  Returns a pointer to currently focused window
+    *
+    *  @return pointer to currently focused window or @c nullptr if there is
+    *          no focused window controlled by the program
+    */
+    UNICORN_EXPORT system::Window* GetFocusedWindow() const;
+
+    /** @brief  Returns the list of known monitors */
+    UNICORN_EXPORT const std::vector<system::Monitor*>& GetMonitors() const;
+
+    /** @brief  Returns monitor associated with given @p window
+    *
+    *  @param  window  window object
+    *
+    *  @return a pointer to monitor object associated with given @p window
+    *          or @c nullptr if something went wrong or the window is not in
+    *          fullscreen mode
+    */
+    UNICORN_EXPORT system::Monitor* GetWindowMonitor(const system::Window& window) const;
+
+    /** @brief  Sets monitor to be used by @p  window for fullscreen mode
+    *
+    *  @param  window      window
+    *  @param  pMonitor    monitor object or @c nullptr for windowed mode
+    *  @param  position    desired position as (x, y)
+    *  @param  size        desired size as (width, height)
+    *  @param  refreshRate desired refresh rate
+    */
+    UNICORN_EXPORT void SetWindowMonitor(const system::Window& window,
+                                         system::Monitor* pMonitor,
+                                         std::pair<int32_t, int32_t> position,
```

Comment:
Maybe pass by const reference, or pass by value, but change parameters to be pure int32_t?